cctp_primitives/src/proving_system/mod.rs

```diff
@@ -0,0 +1,118 @@
+pub mod init;
+pub mod verifier;
+pub mod error;
+
+use crate::{
+    type_mapping::{
+        CoboundaryMarlin, CoboundaryMarlinProverKey, CoboundaryMarlinVerifierKey, CoboundaryMarlinProof,
+        Darlin, DarlinProverKey, DarlinVerifierKey, DarlinProof,
+        FieldElement,
+    },
+    proving_system::{
+        init::{
+            get_g1_committer_key, get_g2_committer_key,
+        },
+        error::ProvingSystemError,
+    },
+    utils::proof_system::ProvingSystemUtils,
+};
+use r1cs_core::ConstraintSynthesizer;
+use rand::RngCore;
+use proof_systems::darlin::pcd::simple_marlin::MarlinProof;
+
+// *************************COBOUNDARY MARLIN********************************
+
+impl ProvingSystemUtils<FieldElement> for CoboundaryMarlin {
+    type Proof = CoboundaryMarlinProof;
+    type ProverKey = CoboundaryMarlinProverKey;
+    type VerifierKey = CoboundaryMarlinVerifierKey;
+
+    fn setup<C: ConstraintSynthesizer<FieldElement>>(
+        circuit: C
+    ) -> Result<(Self::ProverKey, Self::VerifierKey), ProvingSystemError>
```

Comment:
Just an idea: may be we should add to the type_mapping some named structure instead of a simple tuple for the prover and verifier keys? Something more semantically clear, than just an agreement that var.0 - prover key, var.1 - verifier key

Reply:
We can do it, but I like the idea that someone just calls:

`let (pk, vk) = setup(...)`